nav2_mppi_controller/src/optimizer.cpp

```diff
@@ -306,114 +298,135 @@ void Optimizer::propagateStateVelocitiesFromInitials(
 }
 
 void Optimizer::integrateStateVelocities(
-  xt::xtensor<float, 2> & trajectory,
-  const xt::xtensor<float, 2> & sequence) const
+  Eigen::Array<float, Eigen::Dynamic, 3> & trajectory,
+  const Eigen::ArrayXXf & sequence) const
 {
   float initial_yaw = static_cast<float>(tf2::getYaw(state_.pose.pose.orientation));
 
-  const auto vx = xt::view(sequence, xt::all(), 0);
-  const auto wz = xt::view(sequence, xt::all(), 1);
+  const auto vx = sequence.col(0);
+  const auto wz = sequence.col(1);
 
-  auto traj_x = xt::view(trajectory, xt::all(), 0);
-  auto traj_y = xt::view(trajectory, xt::all(), 1);
-  auto traj_yaws = xt::view(trajectory, xt::all(), 2);
+  auto traj_x = trajectory.col(0);
+  auto traj_y = trajectory.col(1);
+  auto traj_yaws = trajectory.col(2);
 
-  xt::noalias(traj_yaws) = xt::cumsum(wz * settings_.model_dt, 0) + initial_yaw;
+  size_t n_size = traj_yaws.size();
+
+  traj_yaws(0) = wz(0) * settings_.model_dt + initial_yaw;
+  float last_yaw = traj_yaws(0);
```

Comment:
Would you agree this is the same and more concise?

```
  float last_yaw = initial_yaw;
  for(size_t i = 0; i != n_size; i++) {
    float & curr_yaw = traj_yaws(i);
    curr_yaw = last_yaw + wz(i) * settings_.model_dt;
    last_yaw = curr_yaw;
  }
```

If so, this and the X/Y entries below it could be updated as such :-) 

The other implementation this function could also be made similarly if the `last_val` variable is introduced

Reply:
Yes, this isthe  same. I'll update other instances of similar code.

Reply:
Sure thing :-)